cob_twist_controller/include/cob_twist_controller/kinematic_extensions/kinematic_extension_base.h

```diff
@@ -51,7 +51,7 @@ class KinematicExtensionBase
 
         virtual bool initExtension() = 0;
         virtual KDL::Jacobian adjustJacobian(const KDL::Jacobian& jac_chain) = 0;
-        virtual JointStates adjustJointStates(const JointStates& joint_states) = 0;
+        virtual JointStates adjustJointStates(const JointStates& joint_states, const geometry_msgs::Pose& pose, const KDL::Twist& twist) = 0;
```

Comment:
Definitely not!
If the DOF class needs pose and twist, only that class should have it as a member variable! The base class must not be touched! Revert that change of the constructor.
Pose ans twist could be set via a set () function of the DOF class..